lib/NodeDomain.js

```diff
@@ -138,14 +148,34 @@ define(function (require, exports, module) {
     NodeDomain.prototype._load = function () {
         if (!this._domainPath) {
             // if path not provided, validate that the domain is already loaded into the connection
-            if (!this.connection.domains[this._domainName]) {
-                var reason = "Domain '" + this._domainName + "' does not exist, nor was a domain path provided";
-                return Promise.reject(reason);
-            } else {
-                this._domainLoaded = true;
-                this._connectionPromise = null;
-                return Promise.resolve();
-            }
+            // If it isn't immediately there, wait a bit before trying again
+            // Because the websocket starts accepting requests before the domain may have been fully loaded
+
+            var domainName = this._domainName,
+                connection = this.connection;
+
+            var validateDomainPromise = new Promise(function (resolve) {
```

Comment:
Fantastic catch! :tropical_fish: 
I think I got confused with the `refreshInterface` in NodeConnection, and thought that since the node was already loaded, that it didn't need to happen again.  Also, since I'm not using any events yet, I didn't notice it.  But I threw in a test event and verified that your hunch was correct; it is necessary.